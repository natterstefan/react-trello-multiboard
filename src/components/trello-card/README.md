# Trello Card

## Iframe Embedding

### Documentation

* https://developers.trello.com/docs/cards

### Issues

When using the [embed script from Trello](https://p.trellocdn.com/embed.js) the
initial loading works fine, but because the embed.js polls in a certain interval
it does not w/o using the onLoad and onResize option. Because their script checks
if the card is still visible. As one can selected all or preferred members, they
might not. Trello will then set the card's iframe height to 0. Adjusting the iframe's
and its parent's css solves the issue. The cards are instantly available when
changing a filter.

The issue is based on the following feature of the embed.js:

```js
// Trello checks if each element is still visible for the user
var elInViewport = function elInViewport(el) {
  if (!document.body.contains(el)) {
    return false
  }
  var bounds = el.getBoundingClientRect()

  return (
    bounds.top <= window.innerHeight &&
    bounds.top + bounds.height >= 0 &&
    bounds.left <= window.innerWidth &&
    bounds.left + bounds.width >= 0
  )
}

// and then later when polling is triggered
togglePolling(card.el, event.origin, event.data.secret, elInViewport(card.el))
```

### Alternative Rendering w/ iframes

One can also create the iframe manually, but resizing (CORS) is then still
an issue.

Example Code:

```js
render () {
  const { card } = this.props
  return (
    <iframe src={`https://trello.com/embed/card?id=${card.id}&secret=${card.id}&compact=true`}/>
  )
}
```

## Solutions from Trello

Solution 1:

```jsx
componentDidMount () {
  window.TrelloCards.create(card.id, $('#card_' + card.id)[0], {
    compact: true,
    onLoad: () => {},
    onResize: () => {}
  })
}

render () {
  return <span className='card_123' />
}
```

Solution 2:

```jsx
componentDidMount () {
  window.TrelloCards.load(document, { compact: true, allAnchors: false })
}

render () {
  const { card } = this.props

  return (
    <blockquote className="trello-card-compact">
      <a href={card.shortUrl}>{card.name}</a>
    </blockquote>
  )
}
```
