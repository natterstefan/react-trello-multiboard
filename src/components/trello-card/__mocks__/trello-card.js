import { mockExampleUser1, mockExampleUser2 } from '../../../__mocks__/mocks'

export const props = {
  addEstimations: jest.fn(),
  boardName: 'example-board',
  card: {
    badges: {
      description: false,
      comments: 1,
      attachments: 1,
      checkItems: 2,
      checkItemsChecked: 1,
    },
    id: 'id-1',
    idBoard: 'board-1',
    idMembers: ['member-1', 'member-2'],
    labels: [
      {
        color: 'orange',
        id: 'label-1',
        name: 'orange',
      },
    ],
    members: [mockExampleUser1, mockExampleUser2],
    name: 'example-name',
    shortUrl: 'https://trello.com/id-1',
  },
  config: {
    estimates_with_round_brackets: false,
    estimates_with_square_brackets: false,
  },
  isHidden: false,
  listName: 'example-list',
  minimizeLabels: true,
  toggleMinimizeLabels: jest.fn(),
}

export const storeStateMock = {
  boards: {
    data: [
      {
        board: {
          // will be added to the TrelloCard (because it matches the idBoards above)
          id: 'board-1',
          name: 'example-board',
        },
      },
      {
        board: {
          id: 'board-2',
          name: 'another-example-board',
        },
      },
    ],
  },
  app: {
    memberToggle: {
      togglePreferred: false,
      togglePreferredMember: null,
    },
    trelloCardUI: {
      minimizeLabels: true,
    },
  },
  members: {
    members: [],
  },
}

export const storeStateMockPreferred = {
  app: {
    memberToggle: {
      togglePreferred: true,
      togglePreferredMember: null,
    },
  },
  members: {
    members: [{ id: 'member-1' }, { id: 'member-2' }],
  },
}

export const storeStateMockPreferredMember = {
  app: {
    memberToggle: {
      togglePreferred: true,
      togglePreferredMember: 'member-1',
    },
  },
  members: {
    members: [{ id: 'member-1' }, { id: 'member-2' }],
  },
}

export const storeStateMockPreferredMemberInvalid = {
  app: {
    memberToggle: {
      togglePreferred: true,
      togglePreferredMember: 'member-3',
    },
  },
  members: {
    members: [{ id: 'member-1' }, { id: 'member-2' }],
  },
}
