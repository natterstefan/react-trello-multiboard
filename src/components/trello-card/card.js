// https://codepen.io/natterstefan/pen/QxJZzW?editors=0110
import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'
import { get, map, includes } from 'lodash'
import FontAwesomeIcon from '@fortawesome/react-fontawesome'

// clicks on the label should not open a new tab with the TrelloCard
const onClick = (event, callback = () => {}) => {
  const classes = get(event, 'target.classList.value') || ''
  if (includes(classes, 'labels')) {
    event.preventDefault()
    callback()
  }
}

// TODO
// - handle case when avatarHash is null, use `initials` on user instead
const TrelloCardUI = ({
  badges,
  boardName,
  labels,
  members,
  minimizeLabels,
  name,
  shortUrl,
  listName,
  toggleMinimizeLabels,
}) => (
  <a
    className="trello-card-wrapper"
    href={shortUrl}
    onClick={event => onClick(event, toggleMinimizeLabels)}
    target="_blank"
  >
    <div className="trello-card">
      <div className="trello-card__content">
        <div className="trello-card--labels">
          {map(labels, label => (
            <span
              key={label.id}
              className={classNames(
                `trello-card--labels-text trello-card--labels-text__${label.color}`,
                {
                  'trello-card--labels-text__minimize': minimizeLabels,
                },
              )}
              title={label.name}
            >
              {label.name}
            </span>
          ))}
        </div>
        <div className="trello-card__title">{name}</div>
        <div className="trello-card__badges">
          {badges &&
            badges.description && (
              <div>
                <FontAwesomeIcon icon={['fa', 'align-left']} size="sm" />
              </div>
            )}
          {badges &&
            badges.comments > 0 && (
              <div>
                <FontAwesomeIcon icon={['fa', 'comment-dots']} size="sm" /> {badges.comments}
              </div>
            )}
          {badges &&
            badges.attachments > 0 && (
              <div>
                <FontAwesomeIcon icon={['fa', 'paperclip']} size="sm" /> {badges.attachments}
              </div>
            )}
          {badges &&
            badges.checkItems > 0 && (
              <div>
                <FontAwesomeIcon icon={['fa', 'check-square']} size="sm" />{' '}
                {badges.checkItemsChecked}/{badges.checkItems}
              </div>
            )}
        </div>
        <div className="trello-card__members">
          <ul>
            {map(members, ({ avatarHash, fullName, initials }) => (
              <li key={`${shortUrl}-${avatarHash}`}>
                <div className="trello-card__member">
                  {avatarHash ? (
                    <img
                      className="trello-card__member--image"
                      src={`https://trello-avatars.s3.amazonaws.com/${avatarHash}/50.png`}
                      alt={fullName}
                    />
                  ) : (
                    <span className="trello-card__member--intials">{initials}</span>
                  )}
                </div>
              </li>
            ))}
          </ul>
        </div>
      </div>
      <div className="trello-card__board-details">
        <div className="trello-card__board-name">{boardName}:</div>
        <div className="trello-card__list-name">{listName}</div>
      </div>
    </div>
  </a>
)

TrelloCardUI.displayName = 'TrelloCardUI'
TrelloCardUI.propTypes = {
  badges: PropTypes.shape({
    description: PropTypes.bool,
    comments: PropTypes.number,
    attachments: PropTypes.number,
    checkItems: PropTypes.number,
    checkItemsChecked: PropTypes.number,
  }).isRequired,
  boardName: PropTypes.string.isRequired,
  labels: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string,
      name: PropTypes.string,
    }),
  ).isRequired,
  listName: PropTypes.string.isRequired,
  members: PropTypes.arrayOf(
    PropTypes.shape({
      avatarHash: PropTypes.string,
      fullName: PropTypes.string,
      initials: PropTypes.string,
    }),
  ).isRequired,
  minimizeLabels: PropTypes.bool,
  name: PropTypes.string.isRequired,
  shortUrl: PropTypes.string.isRequired,
  toggleMinimizeLabels: PropTypes.func,
}

TrelloCardUI.defaultProps = {
  minimizeLabels: false,
  toggleMinimizeLabels: () => {},
}

export default TrelloCardUI
