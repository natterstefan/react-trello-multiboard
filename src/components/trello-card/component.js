import React from 'react'
import { get, invoke, map } from 'lodash'
import proptypes from './prop-types'

import TrelloCardUI from './card'

class TrelloCard extends React.Component {
  componentDidMount() {
    const { config } = this.props
    // Note: previously we added all members (or only preferred) ones from the
    // card to the store. One can add it here again for instance.

    // update the estimations
    const estimation = {
      estimated: config.estimates_with_round_brackets ? this.getEstimation(/\([0-9]+\)/g) : 0, // (%s)
      consumed: config.estimates_with_square_brackets ? this.getEstimation(/\[[0-9]+\]/g) : 0, // [%s]
    }

    invoke(this.props, 'addEstimations', estimation)
  }

  getEstimation(reg) {
    const { card } = this.props
    const cardName = get(card, 'name') || ''

    const estimates = cardName.match(reg)
    if (!estimates) {
      return 0
    }
    let complexity = 0
    for (let i = 0; i < estimates.length; i++) {
      complexity += parseInt(estimates[i].substr(1), 10)
    }
    return complexity
  }

  getMemberClass() {
    const { card } = this.props
    // add an extra class, to make cards selectable via css
    const classes = map(get(card, 'idMembers', []), id => `member_${id}`)
    return classes.join(' ')
  }

  render() {
    const { boardName, card, isHidden, listName } = this.props

    return (
      <div
        id={`card_${card.id}`}
        className={`card ${this.getMemberClass()}`}
        style={{
          display: isHidden ? 'none' : 'block',
        }}
      >
        <TrelloCardUI {...card} listName={listName} boardName={boardName} />
      </div>
    )
  }
}

TrelloCard.propTypes = proptypes
export default TrelloCard
