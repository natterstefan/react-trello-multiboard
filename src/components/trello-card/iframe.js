// NOTE (201804)
// This component is not used anymore, and was replaced with ./card.js. But for
// the record it is still present in the repository. It might be removed in the
// future though.
//
// Note: the following scripts need to be added to index.html if this is enabled
// again:
// - <script src="https://p.trellocdn.com/embed.min.js"></script
// - <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.3/jquery.min.js"></script>

/* global $ */
import React from 'react'
import PropTypes from 'prop-types'
import { get } from 'lodash'

const heightOffset = 10

// usage: <TrelloCardIframe cardId={card.id} />
export default class TrelloCardIframe extends React.Component {
  static propTypes = {
    cardId: PropTypes.string.isRequired,
  }

  constructor(props) {
    super(props)
    this.state = {
      height: 0,
    }

    this.onLoad = this.onLoad.bind(this)
    this.onResize = this.onResize.bind(this)
  }

  componentDidMount() {
    this.createCard()
  }

  onLoad(evt) {
    // Callback after the card has loaded
    // Can be used to resize the parent container
    const clientHeight = evt && get(evt, 'path[0].clientHeight', 0)
    if (clientHeight > 0) {
      this.setState({ height: clientHeight + heightOffset })
      return true
    }
    return false
  }

  onResize(dim) {
    // Callback after the card resizes when the comments
    // section is expanded or collapsed
    if (dim && dim.height > 0) {
      this.setState({ height: dim.height + heightOffset })
      return true
    }
    return false
  }

  createCard() {
    const { cardId } = this.props
    const cardElement = get($(`#card_${cardId}`), '[0]')

    // window.TrelloCards is part of trello's embed.min.js
    if (window.TrelloCards && window.TrelloCards.create) {
      window.TrelloCards.create(cardId, cardElement, {
        compact: true,
        onLoad: this.onLoad,
        onResize: this.onResize,
      })
      return true
    }
    return false

    // NOTE: each iframe also pushes it's new size to the parent site as a message
    // one can subscribe and do something with the data as well. The preferred way
    // is listed and used above though (see onLoad and onResize).
    // var receiveMessage = (event) => {
    //   if (event.data.command === 'resize' && event.data.secret) {
    //     var newHeight = event.data.options.height
    //     if (newHeight !== undefined && newHeight > 0) {
    //       this.setState({height: newHeight + 20})
    //     }
    //   }
    // }
    // window.addEventListener('message', receiveMessage, false)
  }

  render() {
    const { height } = this.state
    const { cardId } = this.props
    /**
     * NOTE: Alternative Solution (eg. when the Trello embed.min.js solution does
     * not work anymore).This solution could still be improved, with eg onLoad
     * --> https://stackoverflow.com/a/3142939/1238150
     */
    // const classes = []
    // each(get(card, 'idMembers', []), id => { if (id) { classes.push(`member_${id}`) } })
    // return (
    //   <iframe
    //     className={`card card_${cardId} ${classes.join(' ')}`}
    //     style={{
    //       border: 0,
    //       boxSizing: 'border-box',
    //       marginBottom: 5
    //     }}
    //     src={`https://trello.com/embed/card?id=${cardId}&secret=${cardId}&compact=true`}
    //   />
    // )
    return (
      <div
        id={`card_${cardId}`}
        style={{
          marginBottom: 15,
          height,
        }}
      />
    )
  }
}

TrelloCardIframe.displayName = 'TrelloCardIframe'
