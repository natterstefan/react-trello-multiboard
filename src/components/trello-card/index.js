import { connect } from 'react-redux'
import { find, get } from 'lodash'
import Card from './component'
import { isCardHidden } from './utils'
import { addMembersEstimation } from '../../actions/members'
import { addBoardEstimations } from '../../actions/boards'

const mapStateToProps = state => ({
  boardData: get(state, 'boards.data', []),
  memberToggle: get(state, 'app.memberToggle', {}),
  members: get(state, 'members.members', []),
})

const mapDispatchToProps = dispatch => ({ dispatch })

const mergeProps = (stateProps, { dispatch }, ownProps) => {
  const addEstimations = estimations => {
    dispatch(addMembersEstimation(ownProps.card.idMembers, estimations, ownProps.card.idBoard))
    // sums total, not only for preferred member estimations
    dispatch(addBoardEstimations(ownProps.card.idBoard, estimations))
  }

  const isCardHiddenProps = {
    cardMembers: get(ownProps, 'card.idMembers', []),
    memberToggle: get(stateProps, 'memberToggle'),
    members: get(stateProps, 'members', []),
  }

  const currentBoard = find(
    get(stateProps, 'boardData'),
    board => get(board, 'board.id') === get(ownProps, 'card.idBoard'),
  )
  const boardName = get(currentBoard, 'board.name', '')

  return {
    ...ownProps,
    boardName,
    isHidden: isCardHidden(isCardHiddenProps),
    addEstimations,
    toggleList: stateProps.toggleList,
  }
}

const CardContainer = connect(mapStateToProps, mapDispatchToProps, mergeProps)(Card)
export default CardContainer
