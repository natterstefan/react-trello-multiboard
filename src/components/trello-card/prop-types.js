import PropTypes from 'prop-types'

export const card = PropTypes.shape({
  id: PropTypes.string.isRequired,
  idMembers: PropTypes.arrayOf(PropTypes.string).isRequired,
  name: PropTypes.string.isRequired,
})

export default {
  addEstimations: PropTypes.func.isRequired,
  boardName: PropTypes.string.isRequired, // Note: currently addded via container
  card,
  config: PropTypes.shape({
    estimates_with_round_brackets: PropTypes.bool,
    estimates_with_square_brackets: PropTypes.bool,
  }),
  listName: PropTypes.string.isRequired,
  isHidden: PropTypes.bool,
  minimizeLabels: PropTypes.bool,
  toggleMinimizeLabels: PropTypes.func,
}
