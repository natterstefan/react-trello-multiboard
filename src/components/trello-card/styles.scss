// https://codepen.io/natterstefan/pen/QxJZzW?editors=0110
.trello-card-wrapper {
  color: #4d4d4d;
  text-decoration: none;
}

.trello-card {
  background-color: rgb(0, 121, 191);
  border-radius: 4px;

  /* Trello css */
  color: #4d4d4d;
  font: 14px 'Helvetica Neue', Arial, Helvetica, sans-serif;
  line-height: 18px;
  margin: 0 auto 15px;
  // max-width: 316px;
  // min-width: 240px;
  padding: 8px;
  width: 316px;

  &--labels {
    margin-top: -10px; // will create a bigger clickable area
    overflow: auto;
    padding: 10px 0 5px; // will create a bigger clickable area
    position: relative;
  }

  &--labels-text {
    @mixin background-color($bgcolor) {
      background-color: $bgcolor;
      color: #fff;
    }

    background-color: #efefef;
    border-radius: 4px;
    color: #000;
    display: block;
    float: left;
    font-size: 12px;
    -webkit-font-smoothing: antialiased;
    font-weight: 600;
    height: 22px;
    line-height: 15px;
    margin: 0 4px 4px 0;
    max-width: 198px;
    min-width: 40px;
    overflow: hidden;
    padding: 3px 6px;
    position: relative;
    text-overflow: ellipsis;
    text-shadow: none;
    white-space: nowrap;
    width: auto;

    &__minimize {
      height: 5px;
    }

    &__green {
      @include background-color(#61bd4f);
    }

    &__yellow {
      @include background-color(#f2d600);
    }

    &__lime {
      @include background-color(#51e898);
    }

    &__orange {
      @include background-color(#ff9f1a);
    }

    &__red {
      @include background-color(#eb5a46);
    }

    &__purple {
      @include background-color(#c377e0);
    }

    &__blue {
      @include background-color(#0079bf);
    }

    &__pink {
      @include background-color(#ff78cb);
    }

    &__black {
      @include background-color(#4d4d4d);
    }
  }

  &__content {
    background-color: #fff;
    border-radius: 3px;
    clear: both;
    display: block;
    min-height: 20px;
    overflow: hidden;
    padding: 10px 10px 5px;
    word-wrap: break-word;
  }

  &__title {
    color: #4d4d4d;
  }

  &__badges {
    // clear: both;
    display: flex;
    float: left;
    margin-right: 15px; // will put members into 2nd row
    margin-top: 5px;
    max-width: 100%;

    div:nth-child(-n + 3) {
      // https://stackoverflow.com/a/8821370/1238150
      margin-right: 10px;
    }

    svg {
      color: #8c8c8c;
      height: 12px;
      margin-bottom: 1px;
    }
  }

  &__members {
    float: right;
    // clear: both;
    margin-top: 5px;

    ul {
      display: flex;
      flex-wrap: wrap;
      list-style-type: none;
      margin: 0;
      padding-left: 0;
    }
  }

  &__member {
    border-radius: 16px;
    display: block;
    float: left;
    height: 32px;
    margin: 0 4px 4px 0;
    overflow: hidden;
    position: relative;
    user-select: none; // no text selection allowed
    width: 32px;

    &--image {
      height: 100%;
      object-fit: cover;
      text-align: center;
      width: 100%;
    }

    &--intials {
      background-color: #e6e6e6;
      display: block;
      font-size: 12px;
      font-weight: bold;
      height: 100%;
      line-height: 30px;
      overflow: hidden;
      text-align: center;
      width: 100%;
    }
  }

  &__board-details {
    align-items: center;
    color: #fff;
    display: flex;
    flex: 1;
    flex-direction: row;
    flex-wrap: wrap;
    margin: 0;
    padding-top: 8px;
  }

  &__board-name {
    font-weight: 700;
    margin-right: 5px;
    overflow: hidden;
    text-overflow: ellipsis;
    white-space: nowrap;
  }

  &__list-name {
    max-width: 230px;
    overflow: hidden;
    text-overflow: ellipsis;
    white-space: nowrap;
  }
}
