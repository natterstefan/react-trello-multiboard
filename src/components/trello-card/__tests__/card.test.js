import React from 'react'
import { shallow } from 'enzyme'

import TrelloCardUi from '../card'

import { props } from '../__mocks__/trello-card'

describe('Component/TrelloCardUi', () => {
  test('should render without throwing an error', () => {
    expect(
      shallow(
        <TrelloCardUi
          {...props.card}
          boardName={props.boardName}
          listName={props.listName}
          minimizeLabels
        />,
      ),
    ).toMatchSnapshot()
  })
})
