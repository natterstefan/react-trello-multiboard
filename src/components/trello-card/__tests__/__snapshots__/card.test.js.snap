// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component/TrelloCardUi should render without throwing an error 1`] = `
<a
  className="trello-card-wrapper"
  href="https://trello.com/id-1"
  onClick={[Function]}
  target="_blank"
>
  <div
    className="trello-card"
  >
    <div
      className="trello-card__content"
    >
      <div
        className="trello-card--labels"
      >
        <span
          className="trello-card--labels-text trello-card--labels-text__orange trello-card--labels-text__minimize"
          key="label-1"
          title="orange"
        >
          orange
        </span>
      </div>
      <div
        className="trello-card__title"
      >
        example-name
      </div>
      <div
        className="trello-card__badges"
      >
        <div>
          <FontAwesomeIcon
            border={false}
            className=""
            fixedWidth={false}
            flip={null}
            icon={
              Array [
                "fa",
                "comment-dots",
              ]
            }
            listItem={false}
            mask={null}
            name=""
            pull={null}
            pulse={false}
            rotation={null}
            size="sm"
            spin={false}
            symbol={false}
            transform={null}
          />
           
          1
        </div>
        <div>
          <FontAwesomeIcon
            border={false}
            className=""
            fixedWidth={false}
            flip={null}
            icon={
              Array [
                "fa",
                "paperclip",
              ]
            }
            listItem={false}
            mask={null}
            name=""
            pull={null}
            pulse={false}
            rotation={null}
            size="sm"
            spin={false}
            symbol={false}
            transform={null}
          />
           
          1
        </div>
        <div>
          <FontAwesomeIcon
            border={false}
            className=""
            fixedWidth={false}
            flip={null}
            icon={
              Array [
                "fa",
                "check-square",
              ]
            }
            listItem={false}
            mask={null}
            name=""
            pull={null}
            pulse={false}
            rotation={null}
            size="sm"
            spin={false}
            symbol={false}
            transform={null}
          />
           
          1
          /
          2
        </div>
      </div>
      <div
        className="trello-card__members"
      >
        <ul>
          <li
            key="https://trello.com/id-1-123456789"
          >
            <div
              className="trello-card__member"
            >
              <img
                alt="Example Username"
                className="trello-card__member--image"
                src="https://trello-avatars.s3.amazonaws.com/123456789/50.png"
              />
            </div>
          </li>
          <li
            key="https://trello.com/id-1-987654321"
          >
            <div
              className="trello-card__member"
            >
              <img
                alt="Another User"
                className="trello-card__member--image"
                src="https://trello-avatars.s3.amazonaws.com/987654321/50.png"
              />
            </div>
          </li>
        </ul>
      </div>
    </div>
    <div
      className="trello-card__board-details"
    >
      <div
        className="trello-card__board-name"
      >
        example-board
        :
      </div>
      <div
        className="trello-card__list-name"
      >
        example-list
      </div>
    </div>
  </div>
</a>
`;
